billing/cap-billing-walkthrough.md

```diff
@@ -0,0 +1,156 @@
+# Automating cost controls by capping billing
+
+## Overview
+
+In this tutorial, you will learn to automate cost controls by setting up programmatic budget notifications. 
+Programmatic budget notifications can be used to disable billing, which will stop the usage of paid services for your project. 
+
+You may choose to disable billing if you have a hard limit on how much money you can spend on Google Cloud. 
+This may be the case for students, researchers, or developers working in sandbox environments.
+
+As you complete this guide, you will learn the following skills: 
++ Creating a budget
++ Setting up a Pub/Sub topic
++ Connecting a billing account to a Pub/Sub topic
++ Deploying a function
+
+**Time to complete:** About 10 minutes
+
+## Getting started 
+
+### Before you begin
+
+Before you attempt this tutorial, you will need:
++ An [active Cloud billing account](https://cloud.google.com/billing/docs/how-to/manage-billing-account#create_a_new_billing_account), where you are a billing admin, or you have been granted the correct level of permissions to complete the steps in this tutorial.
+
+## Understand best practices
+
+We recommend that you configure a separate, single Google Cloud project to contain all of your billing administration needs, including your Cloud Billing-related Pub/Sub topics. Your billing administration Google Cloud project can also be used for things like Cloud Billing Budget API access, Cloud Billing Account API access, Cloud Billing exported data, and so on.
+
+## Select a test project 
+
+For this tutorial, select or create a test project. The function will be acting on this project, not the billing administration project.  
+
+**Caution:** Using the cap billing example will remove Cloud Billing from your project, shutting down all resources. This may result in resources being irretrievably deleted, with no option to recover services. You can re-enable Cloud Billing, but there is no guarantee of service recovery and manual configuration is required.
+
+<walkthrough-project-setup></walkthrough-project-setup> 
+
+```sh
+export GCP_PROJECT={{project_id}}
```

Comment:
Please use `GOOGLE_CLOUD_PROJECT` - this is the env var that's used by the auth library https://googleapis.dev/python/google-auth/latest/reference/google.auth.environment_vars.html?highlight=google_cloud_project It's also automatically populated in our test setup.